walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      //Return the times that a meeting can happen today
+      //MeetingRequest holds all the information regarding atendees, and length of meeting
+      //Event has a set of attendees and a specific timerange the atendees are meeting 
+      //TimeRange is a specific set of time a meeting can start and end
+    Collection<TimeRange> slots = new HashSet<TimeRange>();
+    if (request.getDuration() < TimeRange.WHOLE_DAY.duration()) {
+        slots.add(TimeRange.WHOLE_DAY); //Initializing base case
+    }
+
+    for (Event event : events) {
+        for (String attendee : event.getAttendees()) {
+            if (request.getAttendees().contains(attendee)) {
+                Collection<TimeRange> temp = new HashSet<TimeRange>(slots);
+
+                for (TimeRange slot : temp) {
+                    if (slot.overlaps(event.getWhen())) {
```

Comment:
For code like this when you have something of the structure

```
if (someCondition) {
   // Do a lot of things in here
  ...
}
```

Consider flipping the if condition to a `guard` structure. While `guard` isn't a keyword in Java, the idea is the same as it is in languages that have a `guard` construct, like Swift.

This code would then become:
```
if (!someCondition) {
   return; // or break; or continue (continue in the case of your code);
}
// Do a lot of things, with less indentation...
```